tests/functional/Combobox.js

```diff
@@ -857,6 +857,151 @@ define([
+				assert.isNull(navigatedDescendant, "navigatedDescendant should be still null.");
+			})
+			.pressKeys(keys.ARROW_DOWN)
+			.sleep(500)
+			.execute(executeExpr)
+			.then(function (navigatedDescendant) {
+				assert(/^France/.test(navigatedDescendant),
+					"navigatedDescendant after first ARROW_DOWN: " + navigatedDescendant);
+			})
+			.pressKeys(keys.ARROW_DOWN)
+			.pressKeys(keys.ARROW_DOWN)
+			.pressKeys(keys.ARROW_DOWN)
+			.pressKeys(keys.ARROW_DOWN) // focus on Cananda
+			.sleep(500)
+			.execute(executeExpr)
+			.then(function (navigatedDescendant) {
+				assert(/^Canada/.test(navigatedDescendant),
+					"navigatedDescendant after forth ARROW_DOWN: " + navigatedDescendant);
+			})
+			.pressKeys(keys.ESCAPE)
+			.sleep(500) // wait for async closing
+			.execute(executeExpr)
+			.then(function (navigatedDescendant) {
+				// note: even if the popup it's closed, list should maintain its navigatedDescendant set.
+				assert(/^Canada/.test(navigatedDescendant),
+					"navigatedDescendant after closing the popup: " + navigatedDescendant);
+			})
+			.pressKeys(keys.ARROW_DOWN)
+			.sleep(500) // wait for async opening
+			.execute(executeExpr)
+			.then(function (navigatedDescendant) {
+				assert(/^Canada/.test(navigatedDescendant),
+					"navigatedDescendant after opening the popup: " + navigatedDescendant);
+			})
+			.pressKeys(keys.ARROW_DOWN)
+			.pressKeys(keys.ARROW_DOWN) // focus on Cananda
+			.sleep(500)
+			.execute(executeExpr)
+			.then(function (navigatedDescendant) {
+				assert(/^China/.test(navigatedDescendant),
+					"navigatedDescendant after two ARROW_DOWN: " + navigatedDescendant);
+			});
+	};
+
+	var checkNavigatedDescendantWithSelection = function (remote, comboId) {
+		var executeExpr = "return getNavigatedDescendant(\"" + comboId + "\");";
+		return loadFile(remote, "./Combobox-decl.html")
+			.execute(comboId + ".focus();")
+			.pressKeys(keys.ARROW_DOWN)
+			.sleep(500)
+			.execute(executeExpr)
+			.then(function (navigatedDescendant) {
+				assert.isNull(navigatedDescendant, "navigatedDescendant should be still null.");
+			})
+			.pressKeys(keys.ARROW_DOWN)
+			.sleep(500)
+			.execute(executeExpr)
+			.then(function (navigatedDescendant) {
+				assert(/^France/.test(navigatedDescendant),
+					"navigatedDescendant after first ARROW_DOWN: " + navigatedDescendant);
+			})
+			.pressKeys(keys.ARROW_DOWN)
+			.pressKeys(keys.ARROW_DOWN)
+			.findById(comboId + "_item2")
+			.click() // UK selected
+			.end()
+			.sleep(500)
+			.execute(executeExpr)
+			.then(function (navigatedDescendant) {
+				assert(/^UK/.test(navigatedDescendant),
+					"navigatedDescendant after two ARROW_DOWN: " + navigatedDescendant);
+			})
+			.pressKeys(keys.ARROW_DOWN)
+			.pressKeys(keys.ARROW_DOWN)
+			.sleep(500)
+			.execute(executeExpr)
+			.then(function (navigatedDescendant) {
+				assert(/^Canada/.test(navigatedDescendant),
+					"navigatedDescendant after other two ARROW_DOWN: " + navigatedDescendant);
+			})
```

Comment:
`checkNavigatedDescendantWithSelection` takes `combobox3` as widget to test, which is a multiple-choice combobox.
Hence, when user clicks on _UK_ item the popup does not close. 

Regarding the `navigatedDescendant`, yes we should be checking something else (too). i.e. `aria-activedescendant` which at each user interaction is set to the highlighted/focused element of the combobox's list. 
However it may be worth to keep the test on the navigatedDescendant as this ensures that changes into `delite/KeyNav` and/or `deliteful/List` do not break anything.